packages/node-core/src/indexer/unfinalizedBlocks.service.ts

```diff
@@ -89,8 +91,28 @@ export abstract class BaseUnfinalizedBlocksService<B> implements IUnfinalizedBlo
     return this.finalizedHeader.blockHeight;
   }
 
+  async processUnfinalizedBlockHeader(header?: Header): Promise<number | undefined> {
```

Comment:
We can update `processUnfinalizedBlocks` to convert the block to a header then call this method